### ISSUE TITLE
"Lock screen" does not actually lock anything

### ISSUE THREAD

**Issue by aaime** [linuxmint/cinnamon-screensaver#194]
On a Linux Mint 18.1, I lock the screen either using "CTRL-ALT-L" or picking "lock screen" from the cinnamon menu. The lock screen with the clock appears, but at the first mouse movement or click it will unlock and return to the desktop immediately, without asking for a password.

**Comment by D1mz** [comment]
I'm too on Linux Mint 18.1 but unable to reproduce the bug. Maybe some problem in your lock screen configuration.

**Comment by dqgls** [comment]
Some other users have this issue as well:
https://www.reddit.com/r/linuxmint/comments/5kib0l/linux_mint_181_screensaver_lockscreen_issues/

**Comment by chirag7jain** [comment]
I am facing the same issue

**Comment by arteal** [comment]
Same with me.

Running Arch Linux with this version :
```
[arteal@quagmire ~ ]$ cinnamon-screensaver --version
cinnamon-screensaver 3.2.13
```

**Comment by mtwebster** [comment]
If you stop then restart cinnamon-screensaver from a terminal, do you get any errors when this behavior occurs?

If it's running, do:
`cinnamon-screensaver-command -e` or use `killall cinnamon-screensaver` as root, then:

`cinnamon-screensaver --debug`

This will spit out some extra authentication and locking information while it runs.  Note, if you're still running <=3.2.12, debug mode would spit out all auth communication, including plain password strings.  Version 3.2.13 removed that.  Please be sure to redact any occurrence of such, or upgrade to 3.2.13 before testing this, and pasting the output.

Thanks

**Comment by offa** [comment]
The installation of `python-setproctitle` (as described in #193) solved the issue for me.

**Comment by BandonRandon** [comment]
I originally had this issue as well, but it seemed to have been resolved in one of the recent package updates.

**Comment by ACinnamonUserInUK** [comment]
Just now - and perhaps once or twice before - I have had a similar problem. To wit: my laptop had suspended itself; I woke it up; entered the BIOS password and . . no lock screen, no request for a user password. Mostly, though, I _am_ asked for the user password - my point is that sometimes this does not happen, which strikes me as security risk.

Cinnamon 3.2.7.

**Comment by darkpixel** [comment]
Same here.  Killing the screensaver process and restarting it appears to fix it when it starts acting up.

**Comment by affinityv** [comment]
Same problem for me, the screensaver "gets off the bus".... randomly.  So much for using a LUKS encrypted laptop if the screensaver just dies for no good reason.

$ cinnamon-screensaver --version
cinnamon-screensaver 3.2.13

Jul  3 19:09:46 andrewm-Satellite-P50-A cinnamon-session[26218]: WARNING: t+37039.48962s: Detected that screensaver has left the bus

**Comment by tfulmer1** [comment]
I am experiencing this same issue -- Mint 18.2 and 18.3 with cinnamon-screensaver 3.4.1. When I killed cinnamon-screensaver and restarted in debug mode, it correctly prompted me for a password. Before that, issue persisted through system halts and the 18.2 - 18.3 upgrade/restart

**Comment by affinityv** [comment]
Upgraded to 18.2 now (Linux Mint).  Same problem.  I had previously installed the python-setproctitle package even though the python3 version was installed.  It didn't help. Now the older version is required for blueberry.... go figure.

This is ludicrous, FDE is pointless when someone can walk up to a machine and get access just because the screensaver aborted; that person doesn't even need a password at all!

$ cinnamon-screensaver --version
cinnamon-screensaver 3.4.1

$ cinnamon --version
Cinnamon 3.4.3

**Comment by mtwebster** [comment]
It's frustrating to try to fix a bug you can't reproduce as well.  The FDE note is new, however..

Does everyone having this issue use encryption of some sort?

Does this happen in a new user's session of you create one?

You all have to understand, if you don't give me sufficient information to reproduce something, the chances of it getting fixed are pretty low.  The problem is sometimes you guys don't know what info is relevant, and sometimes I just don't ask the right questions ;)

**Comment by ACinnamonUserInUK** [comment]
> Does everyone having this issue use encryption of some sort?

I do _not_ use encryption (I did once try encrypting my home partition and it broke something - I can't remember what - so I had to remove it, sadly). Or I don't unless one counts VeraCrypt.

> Does this happen in a new user's session of you create one?

I am afraid I don't know and I can't find out, because I saw the problem only once, some months ago.

Relevant system information (er, for my system now; I had earlier versions of Cinnamon and the screensaver - respectively - when I had the problem):

```
$ cinnamon-screensaver --version
cinnamon-screensaver 3.4.1

$ cinnamon --version
Cinnamon 3.4.3
```

**Comment by tfulmer1** [comment]
@mtwebster I'm using the mint provided FDE as well. However, I'm having issues with it not always successfully loading the decryption prompt and thus the system not booting so I'm looking into alternatives. If I reset the system to otherwise identical install but no FDE I'll let you know results. I also haven't halted the system since I used the restart option which fixed it for this session, I'll let you know results from that. 

For those coming across this the first time, if the restart fixes your issue you can restart again without debug mode and disown the process to return to normal operation.

I'm watching the thread since I started having the issue, so let me know if you have further questions or want me to test anything.

**Comment by affinityv** [comment]
I wouldn't think that FDE would be relevant.  The common thread is a migration from 18 to 18.1 and now 18.2 for Linux Mint at least.  I can leave my laptop locked, go away and come back later to find it isn't locked and the screensaver has left the bus.

**Comment by aaime** [comment]
In my case (18.1) it's not dying anymore, but every time I unlock I can see the screen contents for a good while (1-2 seconds) before the lock screen appears and I can enter the password. Not good, the screen might contain private information that anyone can have a look at (repeatedly too, if they have some patience).

**Comment by tfulmer1** [comment]
@aaime I don't think that's related to cinnamon-screensaver specifically,  I've had that happen to me when coming back from blank lock screen on Ubuntu. It doesn't happen every time, but enough I've noticed and started trying to track it down. So I think it's an upstream problem with locking

**Comment by mtwebster** [comment]
There was a locking issue fixed in cinnamon-session for 3.4 (mint 18.2) - https://github.com/linuxmint/cinnamon-session/commit/c2d18069e62c331ba65551871d2d15cd27ebbc85.

It wouldn't affect everyone, but if had normal screensaver locking off, but still wanted to lock when suspending or hibernating, this could end up causing a visible desktop briefly when coming back.  The fix ensures the locking occurs fully *prior* to suspend/hibernate.

I also never considered encryption to relevant, but clearly if some people are having issues, we're missing something, so I'm willing to entertain anything that may be 'different' from the norm with users.  I need to be able to reproduce these issues.

**Comment by aaime** [comment]
@mtwebster I just made a quick test now that I've upgraded to 18.2, indeed I could not see the desktop anymore. Just one test though, I'll report back if I can see it again.

**Comment by affinityv** [comment]
I don't know if the debug output would help, but I'm sure it would make a good start; you can find it here:
https://affinityvision.com.au/cinnamon-screensaver--debug.txt

I fiddled around closing my laptop lid, turning on and off the externally attached screen, locking with keyboard shortcut, allowing the lock to happen by itself due to timeout.  All the fiddling didn't break anything, until it finally segfaulted.

If you want me to do anything else, please let me know; right now I am heading off to bed though and I've got plenty on tomorrow, so I don't know when I'll be available next.

**Comment by mtwebster** [comment]
Hmm if I could get a trace where that happens it may shed some light.  It can be a bit tricky to get one, since it's the screensaver, and when it's frozen by a debugger, it can make things difficult to interact with a normal desktop.

If anyone is interested, these steps should allow you to get a trace when the screensaver crashes in a fairly painless manner.  I would suggest a reboot after going through running the script - you get the occasional zombie processes.

And before someone says it, yes we should have an easier way to do this  ;)  - we're working on it.

```
# Download our cin-debug script and make it executable.
# You can just paste this url into a browser and get it that way if you want.

wget https://raw.githubusercontent.com/linuxmint/Cinnamon/master/utils/cin-debug
chmod +x ./cin-debug

# Install a few things...

sudo apt-get install gdb screen cinnamon-dbg
# gdb is the debugger.
# screen lets you have 'virtual' terminals that can be doing something independent of
  any reliance on the current graphical environment.
# cinnamon-dbg brings in debug symbols for pretty much all the relevant libraries
  we use throughout cinnamon


# Open two terminals
# In the first, run:

cinnamon-screensaver-command -e            # kill any existing screensaver
cinnamon-screensaver --debug               # start a new one


# In the second terminal, navigate to the folder where cin-debug is...

# start screen, you might get an intro screen, just continue:

screen

# start our script, which attaches to the running screensaver process
# This requires a password, gdb needs root to attach to running processes

./cin-debug cinnamon-screensaver

# The debugger will start up, ending with <continuing.. 

Ctrl-A, Ctrl-D       # detaches back to your original 'terminal'
```

From here, try to do whatever to get any crash of the screensaver.  If/when it does crash, that debugger instance should save a timestamped file in your home folder, something like:

`cinnamon-screensaver-2017-07-05-9:52:44.log`

Thanks

**Comment by mtwebster** [comment]
Just an update, we found what's likely causing the crash during hotplugging and are working on a patch, hopefully in the next day or so to test out.

I'd still love crash reports for those it may be happening to - I'm pretty sure there is more than one issue here (for different posters)

**Comment by jkirk** [comment]
This might just be another puzzle piece but here the cinnamon-screensaver does not show a password-prompt when the user interface scaling is set to Hi-DPI. Instead the desktop can be seen, the windows can be switched via the taskbar but I can not focus a window. It sees like the desktop is frozen but when entering the password (blindly without a password prompt) the desktop gets unlocked again.

When switching to "normal" user interface scaling the desktop is locked and the password prompt is shown.

The debug output can be found [here](https://gist.github.com/jkirk/9cb469e00cb5ad19b31c39fcd04f407a).

**Comment by mtwebster** [comment]
@jkirk this happens any time the screensaver activates?  Or when returning and 'raising' the unlock dialog?

**Comment by affinityv** [comment]
I think the problem has a lot to do with detecting an external screen; if I completely leave the screen on (don't turn off the power completely), then I'm not seeing the problem.  It may also be related to lid-close as well for the internal screen.

**Comment by jkirk** [comment]
Yes, this only happens when my external (4K) screen is connected.

Setting the user interface scaling to "Hi-DPI" when no external screen is connected, just with my notebook screen (with a resolution of 1366x768) the password prompt is shown (in huge letters).

(I updated my [comment](https://github.com/linuxmint/cinnamon-screensaver/issues/194#issuecomment-315569466))

**Comment by mtwebster** [comment]
This is indeed related to hotplug events/xrandr changes.  The screensaver isn't handling the various events well enough.  This is a priority to fix, I'm just waiting on a piece of hardware to do some more complete tests.

**Comment by mtwebster** [comment]
I've got a patch ready, just waiting for a bit more internal testing.

https://github.com/mtwebster/cinnamon-screensaver/commit/fce51c4cd4f3b5e6fffa54856e4a4c9b1f7346d0

**Comment by jkirk** [comment]
Any (inofficial) deb-package available which we can test?

**Comment by aaime** [comment]
Hey, since a few days ago the screen saver locks... too much! When I get back to it, X server seems to lock solid and I cannot move the mouse anymore. 100% repeatable, I have to pay attention not to lock the screen, or I have to drop on the command line and restart mdm. Probably a separate issue but, wondering, anyone else experiencing the same?

**Comment by affinityv** [comment]
Linux Mint changelog????? Understated or what!
   [ Michael Webster ]
   * add an issue template
   * CsScreen: Fix hotplug/screen change detection and response, other issues.
   * logindClient.py: Produce a more helpful message when logind is running, but we're unable to figure out a valid session id.
   * manager/stage: Allow a second activation trigger to override a previous.
   * manager.py: Force secondary activations to be immediate. (prevents flash)

THIS IS NOT JUST A SOFTWARE UPDATE, IT FIXES A MAJOR VULNERABILITY.

**Comment by explorador-dm** [comment]
I have the same problem: Linux Mint 18.2 Cinnamon 64-bit doesn't prompt for a password after being suspended when I reopen the lid.

I logged cinnamon screensaver logs to the attached file. There are critical errors and failed assertions at the end of the log file.

One more thing: `cinnamon-screensaver --debug &>~/cinnamon-screensaver-debug.log` command ended with a `Segmentation fault`.

Hope this helps to find the problem and the solution.

[cinnamon-screensaver-debug.txt](https://github.com/linuxmint/cinnamon-screensaver/files/1255372/cinnamon-screensaver-debug.txt)

**Comment by Rusty-1** [comment]
Some further info. I find that when I'm working on a dual displays that 1 displays the lock screen and the other is unlocked and I can interact with the "unlocked" with the mouse but all keyboard input is captured by the login screen. I hope that makes sense.

**Comment by jaliss** [comment]
I attached an external monitor while my laptop screen was locked and I got access to the desktop on it. I was able to use the menu/run apps and access the desktop docs while the laptop screen prompted for my password.  I reproduced this twice only.

**Comment by clefebvre** [comment]
https://github.com/linuxmint/cinnamon-screensaver/issues/219

**Comment by johnbburg** [comment]
I am also seeing the issue [described by jkirk](https://github.com/linuxmint/cinnamon-screensaver/issues/194#issuecomment-315569466). I use a vertically oriented second monitor, and I can see what looks like the password screen overlay partially obscuring the second monitor at the top of the screen. I'm unable to perform any keyboard input (seems to go to the obscured password prompt). But I can use the trackpad to view, move, minimize, and close windows. When I unplug the second monitor, the password prompt reverts to the primary screen, and I am able to input my password and unlock the screen.  I took a video, which I'd be happy to share in a PM.

**Comment by christian-schwarzbauer** [comment]
I'm using Cinnamon on Antergos with 2 external monitors over HDMI and experienced the same problem as described by (at least) @johnbburg, @jaliss & @Rusty-1.

I just reproduced it 3 times as follows:
* put laptop without any monitors attached into lock screen
* attach first external monitor via HDMI and wait for it to show
* attach dell USB-C docking station which has second external monitor connected via HDMI
* wake second external monitor from sleep and wait for it to show
* BOOM -> lock screen only covers part of first external monitor! second external monitor and laptop screen are both fully accessible via the mouse (keyboard focus is still in lock screen password field)

this has a *huge* security impact, so any comment from the cinnamon-team would be highly appreciated!

**Comment by mtwebster** [comment]
Can you run this in debug mode and provide the logging when you reproduce it?

```
cinnamon-screensaver-command -e
cinnamon-screensaver --debug
```

**Comment by christian-schwarzbauer** [comment]
sure, here it is:
(in between I removed 383 same lines mentioning the "authClient initialize...")

```
Debug mode active
/usr/share/cinnamon-screensaver/cinnamon-screensaver-main.py:86: Warning: g_base64_encode_step: assertion 'in != NULL' failed
  css = provider.to_string()
Cinnamon Screensaver support not found in current theme - adding some...
Trying to connect to logind...
Starting screensaver...
Not running under the session scope, trying XDG_SESSION_ID
found session: c2
Successfully using logind
couldn't grab keyboard
Found 1 Xinerama screens on display :0
Monitor 0 is 0,0 1920 x 1080
Cinnamon Screensaver compiled without Solaris Xinerama support
Scale factor of 1 applied.  Monitor 0 is 0,0 1920 x 1080
Stage.update_geometry - new backdrop position: 0, 0  new size: 1920 x 1080
Found 2 Xinerama screens on display :0
Monitor 0 is 0,0 3840 x 2160
Monitor 1 is 3840,0 1920 x 1200
Cinnamon Screensaver compiled without Solaris Xinerama support
Scale factor of 1 applied.  Monitor 0 is 0,0 3840 x 2160
Scale factor of 1 applied.  Monitor 1 is 3840,0 1920 x 1200
Stage: Received screen monitors-changed signal, updating monitor views
Stage: Received screen changed signal, updating backdrop
Stage.update_geometry - new backdrop position: 0, 0  new size: 5760 x 2160
Found 2 Xinerama screens on display :0
Monitor 0 is 0,0 3840 x 2160
Monitor 1 is 3840,0 1920 x 1200
Cinnamon Screensaver compiled without Solaris Xinerama support
Scale factor of 2 applied.  Monitor 0 is 0,0 1920 x 1080
Scale factor of 2 applied.  Monitor 1 is 1920,0 960 x 600
Stage: Received screen monitors-changed signal, updating monitor views
Notification received...
Sender: NetworkManager
Summary: Connection Established
Body: You are now connected to “Wired connection 1”.
Found 3 Xinerama screens on display :0
Monitor 0 is 1920,0 1920 x 1200
Monitor 1 is 0,0 1920 x 1200
Monitor 2 is 0,1200 1920 x 1080
Cinnamon Screensaver compiled without Solaris Xinerama support
Scale factor of 2 applied.  Monitor 0 is 960,0 960 x 600
Scale factor of 2 applied.  Monitor 1 is 0,0 960 x 600
Scale factor of 2 applied.  Monitor 2 is 0,600 960 x 540
Stage: Received screen monitors-changed signal, updating monitor views
Stage: Received screen changed signal, updating backdrop
Stage.update_geometry - new backdrop position: 0, 0  new size: 1920 x 1140
Found 3 Xinerama screens on display :0
Monitor 0 is 1920,0 1920 x 1200
Monitor 1 is 0,0 1920 x 1200
Monitor 2 is 0,1200 1920 x 1080
Cinnamon Screensaver compiled without Solaris Xinerama support
Scale factor of 1 applied.  Monitor 0 is 1920,0 1920 x 1200
Scale factor of 1 applied.  Monitor 1 is 0,0 1920 x 1200
Scale factor of 1 applied.  Monitor 2 is 0,1200 1920 x 1080
Stage: Received screen monitors-changed signal, updating monitor views
authClient initialize... initialized already: False
authClient initialize... initialized already: True
authClient initialize... initialized already: True
authClient initialize... initialized already: True
authClient initialize... initialized already: True
authClient initialize... initialized already: True
pam_start ("cinnamon-screensaver", "cschwarzbauer", ...) ==> 0 (Success)
Handling message style 1: 'Password: '
Waiting for respose to message style 1: 'Password: '
Waiting for lock
Waiting for response
Got message style 1: 'Password: '
Output from pam helper: 'CS_PAM_AUTH_SET_PROMPT_Password: _'
authClient idle add auth-prompt
Output from pam helper: 'CS_PAM_AUTH_BUSY_FALSE'
authClient idle add auth-busy
Output from pam helper: ''
authClient initialize... initialized already: True
...
383x same message -> removed
...
authClient initialize... initialized already: True
authClient message to child
authClient initialize... initialized already: True
auth_message_handler processing response string
should interrupt: 0
Got response
Got respose to message style 1: interrupt:0
Msg handler returned 1
Output from pam helper: 'CS_PAM_AUTH_BUSY_TRUE'
authClient idle add auth-busy
Output from pam helper: ''
   pam_authenticate (...) ==> 0 (Success)
pam_acct_mgmt (...) ==> 0 (Success)
   pam_setcred (...) ==> 0 (Success)
 pam_end (...) ==> 0 (Success)
Verify user returned: TRUE
Output from pam helper: 'CS_PAM_AUTH_SUCCESS'
authClient idle add success
Output from pam helper: ''
authClient message to child
authClient helper process completed...
CsScreen dispose
CsScreen finalize
```

**Comment by russmack** [comment]
Currently experiencing this with Tara, cinnamon 3.8.8.  Lock screen variously not doing anything at all, or showing an unlocked lock screen, and suspend suspending without locking.

**Comment by GrangeBeach** [comment]
i am having same experience as russmack above.

for me first time i am experiencing this problem.

is there a way to completely remove cinnamon-screensaver from linux mint 19 and still get screen locking and suspend (all i need) ?

**Comment by mtwebster** [comment]
@christian-schwarzbauer Do you have hidpi scaling set to "Auto"?  (Cinnamon-settings -> general)

It appears that at different times the scaling changes, causing incorrect calculation of the screen size.  I suspect this may be caused by having monitors of varying resolutions, with sleep/wake causing detection to be calculated on different monitors at different times.

Could you set that option to "Normal" or "Double" rather than "Auto"

@GrangeBeach You don't need a screensaver for suspend, but you do for locking.  Could you try the steps [here](https://github.com/linuxmint/cinnamon-screensaver/issues/194#issuecomment-366663747)?

Another test would be to create a new user and see if it functions properly under that user.  These would be a good start for troubleshooting.

Thanks

**Comment by flansuse** [comment]
Forum post on the Mint forums, for reference: https://forums.linuxmint.com/viewtopic.php?f=208&t=274155&p=1502954#p1502954

I've managed to make one of the issues (not properly locking the screen when suspending the computer) **100% reproducible** on **Linux Mint 19 Cinnamon** with the following package versions:

```
cinnamon 3.8.8+tara

cinnamon-screensaver 3.8.2+tara
```


This might not address all the underlying causes, but it's a start! Perhaps the developers can use this to further investigate the code that needs to be rewritten. I will post this finding on the GitHub bug reports.

Here is how I can trigger this issue, and here is the text from **--debug** mode:

```
Calling XResetScreenSaver
manager: user activity, waking
manager: not locked, queueing idle deactivation
CsScreen dispose
CsScreen finalize
couldn't grab keyboard
couldn't grab keyboard
couldn't grab keyboard
couldn't grab keyboard
Nuking focus
couldn't grab keyboard
couldn't grab keyboard
couldn't grab keyboard
couldn't grab keyboard
couldn't grab mouse
couldn't grab mouse
couldn't grab mouse
couldn't grab mouse
Calling XResetScreenSaver
```



### Here is how I discovered a way to 100% reproduce this security issue:

1) Make sure you have it set to lock the screen upon suspend and when the screensaver starts. Settings > Screensaver > Settings

2) Create a shortcut key / key combination for suspending the computer to RAM. If you have it set to suspend upon closing the lid, it will behave the same way if you follow the next steps.

4) Right-click somewhere, such as on the taskbar or a file or the desktop itself, in order to bring up a context menu. Do not do anything else with the mouse. Leave the cursor where it is.

5) Now close the laptop's lid. Wait a while, and you will hear the system go into suspend mode.

6) Now open the laptop's lid and you will hear the system wake up.

**7) You have access to your entire unlocked session, completely bypassing any password dialogue or lock screen!**

This can also be done via the shortcut keys, but it requires a more unlikely (though not improbable) timing of events. I had success exploiting the issue: if your press the shortcut key to suspend, and immediately bring up any context menu (or any such menu to steal the cursor's focus), you can reproduce the issue. However, the timing needs to be precise, so it's less likely to happen, yet it can still happen by mistake.

This still does not explain why there is a general degradation of Cinnamon over time, and why the screen locker gets slower and slower over time. They might be separate issues, or perhaps inter-related with this one. I hope this post helps the developers narrow down their search for the code responsible for this vulnerability.

**Comment by JosephMcc** [comment]
@flansuse This is a known issue. There is at least one other open here about it. I don't remember the specifics but this will happen with any application menu open. It has to do with how focus works and  I'm not sure we can even change it on our side.

https://github.com/linuxmint/cinnamon-screensaver/issues/258
https://github.com/linuxmint/cinnamon-screensaver/issues/93

**Comment by flansuse** [comment]
@JosephMcc Is this unique to Cinnamon, or applies to any desktop?

**Comment by affinityv** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

All I have to do for the screen saver NOT to lock, is to open the
cinamin menu and leave the laptop alone, when I come back, the screen
is blank, but it doesn't lock and no password is required to contniue.

# dpkg-query -l cinna*
Desired=Unknown/Install/Remove/Purge/Hold
|
Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-
pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                         Version
       Architecture                Description
+++-============================================-=======================
====-===========================-=======================================
======================================================
ii  cinnamon                                     3.6.7+sylvia
       amd64                       Modern Linux desktop
un  cinnamon-bluetooth                           <none>
       <none>                      (no description available)
un  cinnamon-capplets-data                       <none>
       <none>                      (no description available)
ii  cinnamon-common                              3.6.7+sylvia
       all                         Cinnamon desktop (Common data files)
ii  cinnamon-control-center                      3.6.5+sylvia
       amd64                       utilities to configure the Cinnamon
desktop
ii  cinnamon-control-center-data                 3.6.5+sylvia
       all                         configuration applets for Cinnamon
- - data files
ii  cinnamon-desktop-data                        3.6.2+sylvia
       all                         Common files for Cinnamon desktop app
s
ii  cinnamon-l10n                                3.6.4
       all                         Translation files for the Cinnamon
desktop
ii  cinnamon-screensaver                         3.6.1+sylvia
       all                         Cinnamon screen saver and locker
ii  cinnamon-screensaver-pam-helper              3.6.1+sylvia
       amd64                       PAM helper for cinnamon-screensaver
un  cinnamon-screensaver-webkit-plugin           <none>
       <none>                      (no description available)
ii  cinnamon-screensaver-x-plugin                3.6.1+sylvia
       all                         XScreensaver plugin for
cinnamon-screensaver
ii  cinnamon-session                             3.6.0+sylvia
       amd64                       Cinnamon Session Manager - Minimal
runtime
ii  cinnamon-session-common                      3.6.0+sylvia
       all                         Cinnamon Session Manager - common fil
es
ii  cinnamon-settings-daemon                     3.6.2+sylvia
       amd64                       daemon handling the Cinnamon
session settings
ii  cinnamon-themes                              2017.06.20
       all                         Cinnamon themes
un  cinnamon-translations                        <none>
       <none>                      (no description available)


Cheers
A.
-----BEGIN PGP SIGNATURE-----

iF4EAREIAAYFAltcTP8ACgkQqBZry7fv4vtQvwEA0ZUOhsJe3n8uSMQkpBmRMe35
SvZuUsUe99uh6heYzKEBANb95LGorJljpfyy8qkf0iQ0mFPEXeGgBpYyAb/YbMLV
=qDDn
-----END PGP SIGNATURE-----

**Comment by torbensky** [comment]
The Mint lock screen reliably shows my desktop and even lets me interact with elements of the desktop when I switch between displays after locking. Is it just using some sort of overlay to block interactions and that's not scaling when the display settings change? Seems really insecure and unusable as a security tool.

**Comment by sruester** [comment]
@mtwebster The problem disappears when setting hidpi scaling from "Auto" to "Normal" or "Double".
My setup is: A laptop with QHD display and an external monitor (via DisplayPort) with HD resolution.
Problem: When I connect the external monitor the lock screen changes its size to 1920x1080 on the QHD display and to 960x540 on the HD monitor. This ends up with the lock screen covering only 1/4 of each screen.

**Comment by Alkaris** [comment]
I'm having this issue here it seems on Linux Mint 19.1

```
]$ cinnamon-screensaver --debug
Debug mode active
/usr/share/cinnamon-screensaver/cinnamon-screensaver-main.py:86: Warning: g_base64_encode_step: assertion 'in != NULL' failed
  css = provider.to_string()
Cinnamon Screensaver support not found in current theme - adding some...
Trying to connect to logind...
org.gnome.ScreenSaver is gone from the session bus
org.mate.ScreenSaver is gone from the session bus
Starting screensaver...
Not running under the session scope, trying XDG_SESSION_ID
found session: c2

(cinnamon-screensaver-main.py:22852): GLib-GIO-WARNING **: 13:15:13.956: Received property Position with type i does not match expected type x in the expected interface
Successfully using logind
```

**Comment by codingthat** [comment]
Having this on Linux Mint 19 MATE just today; had been working fine before that.  I know this is a cinnamon thread but I thought I would post here in case the culprit lies deeper.  My only updates before the latest restart were:

![image](https://user-images.githubusercontent.com/26723818/61697143-fa425080-ad36-11e9-928d-d42abce330f3.png)

**Comment by andromeda306** [comment]
Not sure whether this issue is still occurring for anyone else. But I'm still regularly experiencing problems with the lockscreen. Currently using cinnamon-screensaver 4.2.0, Cinnamon 4.2.4. Basically at what seem like random times I'll open up my laptop from suspended state (overnight) and the lockscreen won't even appear i.e the desktop is made available straight up. I'd say this is a pretty severe security issue and would appreciate any additional tips/questions. Happy to provide any additional  system info if it can help solve the issue asap.

**Comment by codingthat** [comment]
Lately I've had this issue far less often, but it still does happen once in a while, I think when there's some complication like me closing the lid too quickly after hitting the keyboard shortcut to lock the screen.

**Comment by andromeda306** [comment]
Most of the time I work with a usb-c dock for charging and external monitors attached and when I manually lock the screen (windows+L) I don't think there has been any issues. The issue most commonly occurs outside this setup, after I open up the laptop from system suspend at home or the next morning - in both cases prior to attaching external devices (apart from a usb keyboard which I am yet to rule out)

**Comment by McThump** [comment]
Some additional data points...  

One computer here (a Dell OptiPlex 7010 with 2 monitors) is having this problem (timed auto-lock never occurs). The machine is running Cinnamon 3.8.8 on Debian 10.2, cinnamon-screensaver version 3.8.2.  

(1) Restarting the computer resolves the problem for a few hours. 
(2) Disconnecting all peripherals except the monitors (in case something was injecting into a keyboard or mouse port) does not resolve the problem. 
(3) The problem does not occur when using xscreensaver instead of the cinnamon-screensaver.  
(4) Most interesting is that after erasing the hard-drive and installing Windows 10, the problem also occurs with the native Windows screensaver.  As with the Debian/Cinnamon setup, restarting Windows resolves the problem for a few hours. 

Except for this screensaver issue, the computer works perfectly.

**Comment by andromeda306** [comment]
I'm currently running the most recent 19.3 version of Mint and I have to say, I think the problem is even worse than in previous releases. I'd be happy to provide further debug information to someone who is a bit more familiar with the inner workings of cinnamon. However as it stands, I'm going to reinvestigate implementing xscreensaver as I never got around to really fixing the issues lat time around.

**Comment by codingthat** [comment]
That would be fantastic!  I find plugging/unplugging an HDMI cable (especially if the external screen takes over as your primary display whenever plugged in) exacerbates the issue even further.  Hopefully a new implementation would take care of that too...

**Comment by qdinar** [comment]
aaime commented on 5 Jul 2017
> ... every time I unlock I can see the screen contents for a good while (1-2 seconds) before the lock screen appears and I can enter the password. ...

this happened once at my laptop, after waking it up, today. not long before i put it to sleep before this incident, a website, in firefox, used 100% cpu and something were using hdd much, as i heard. also, unusually many tabs were opened in firefox, as i remember.

Package: cinnamon-screensaver
Version: 4.4.1+tricia

Package: cinnamon
Version: 4.4.8+tricia

linux mint 19.3 64 bit.

**Comment by areographe** [comment]
```
Kernel: 5.4.31-1-MANJARO x86_64 bits: 64 Desktop: Cinnamon 4.4.8  Distro: Manjaro Linux 
cinnamon-screensaver 4.4.1
```

just had an issue where cinnamon-screensaver didn't run after rebooting, so pressing 'Lock' did nothing, and at the 15 minute mark (normal time before screensaver starts) the screen just turned off and moving the mouse woke it up again. no lock screen. using alt+f2 to run the command 'cinnamon-screensaver' seems to work now though.

**Comment by areographe** [comment]
just to follow up:

looks like this journalctl output is showing something related to the screensaver being killed before the bug appears  
```

Apr 12 18:14:44 <host> cinnamon-session[1142]:_ WARNING: t+69865.21877s: Application 'cinnamon-settings-daemon-screensaver-proxy.desktop' killed by signal 15
Apr 12 14:34:57 <host> cinnamon-session[1142]: WARNING: t+56678.28012s: Application 'cinnamon-settings-daemon-screensaver-proxy.desktop' killed by signal 1
Apr 11 14:48:29 <host> cinnamon-session[1142]: WARNING: t+10958.45004s: Application 'cinnamon-settings-daemon-screensaver-proxy.desktop' killed by signal 1
```
i'm also getting these messages in journalctl as well:

```
Apr 12 17:15:42 luke-thinkpad cinnamon-session[1142]: WARNING: t+66323.01120s: Detected that screensaver has left the bus
```
prior to my last manjaro update i have 90% of my journalctl messages about the screensaver being detected appearing on the bus and only 10% being leaving the bus, but as of 2 days ago it's about 50-50.

hope this helps narrow it down
